packages/test/test-utils/src/testFluidObjectInternal.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	IFluidHandle,
+	IRequest,
+	IResponse,
+	type IFluidLoadable,
+} from "@fluidframework/core-interfaces";
+import { FluidObjectHandle } from "@fluidframework/datastore/internal";
+import {
+	IChannelFactory,
+	IFluidDataStoreRuntime,
+	type IChannel,
+} from "@fluidframework/datastore-definitions/internal";
+import {
+	IFluidDataStoreChannel,
+	IFluidDataStoreContext,
+} from "@fluidframework/runtime-definitions/internal";
+import { create404Response } from "@fluidframework/runtime-utils/internal";
+import type { ISharedObject } from "@fluidframework/shared-object-base/internal";
+
+/**
+ * A test Fluid object that will create a shared object for each key-value pair in the factoryEntries passed to load.
+ * The shared objects can be retrieved by passing the key of the entry to getSharedObject.
+ * It exposes the IFluidDataStoreContext and IFluidDataStoreRuntime.
+ * @remarks
+ * This is a simplified (does not use SharedMap) alternative to {@link TestFluidObject} which does not implement the external facing {@link ITestFluidObject} interface.
+ * @internal
+ */
+export class TestFluidObjectInternal implements IFluidLoadable {
+	public get IFluidLoadable() {
+		return this;
+	}
+
+	public readonly handle: IFluidHandle<this>;
+	private initializationPromise: Promise<void> | undefined;
+
+	/**
+	 * Creates a new TestFluidObjectInternal.
+	 * @param runtime - The data store runtime.
+	 * @param context - The data store context.
+	 * @param factoryEntries - A list of id to IChannelFactory mapping. For each item in the list,
+	 * a shared object is created which can be retrieved by calling getSharedObject() with the id;
+	 */
+	constructor(
+		public readonly runtime: IFluidDataStoreRuntime,
+		public readonly channel: IFluidDataStoreChannel,
+		public readonly context: IFluidDataStoreContext,
+		private readonly factoryEntriesMap: Map<string, IChannelFactory<ISharedObject>>,
+	) {
+		this.handle = new FluidObjectHandle(this, "", runtime.objectsRoutingContext);
+	}
+
+	/**
+	 * Retrieves a shared object with the given id.
+	 * @param id - The id of the shared object to retrieve.
+	 */
+	public async getSharedObject(id: string): Promise<IChannel> {
```

Comment:
Maybe rename this to something like `getInitialSharedObject` if it's not too much pain to update all current usages.

Reply:
While looking at the usage of this in `tree/src/test/utils.ts`, I got confused for a minute as to where is the creation for this tree it's getting. Then I realized that it was created as part of initialization. Renaming this to reflect that would hopefully clear that confusion also.